entity/option.go

```diff
@@ -0,0 +1,54 @@
+package entity
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/ockam-network/did"
+	"github.com/ockam-network/ockam"
+)
+
+// Option is a functional option that represents
+// an arbitrary entity configuration
+type Option func(*Entity) error
+
+// OptionAttributes is a functional option for configuring
+// entity attributes.
+// TODO - validate attributes, should they be non-zero?
+func OptionAttributes(attributes Attributes) Option {
+	return func(e *Entity) error {
+		if len(attributes) == 0 {
+			fmt.Printf("warn: attributes are empty")
+		}
+
+		e.attributes = attributes
+		return nil
+	}
+}
+
+// OptionSigner is a functional option for setting the entity
+// signer attribute.
+func OptionSigner(s ockam.Signer) Option {
```

Comment:
The "Option-" prefix for options has both pros and cons. 

The main pro is that it makes it clear which functions are functional options ... but I think that point is also communicated by the fact that all such functions return `Option` so it is apparent in their signature and it can the first line of their godoc.

The con is that they make the API noisy ...

```
entity.New(entity.ID(id), entity.Signer(signer))
```
reads a lot better than:

```
entity.New(entity.OptionID(id), entity.OptionSigner(signer))
```
this gets a lot worse if the number of options is more than a few.

This is why we chose to keep the main Entity interface implementation in its own package `ockam/entity` and not have the Option prefix